src/scripts/validation.js

```diff
@@ -0,0 +1,81 @@
+export const settings = {
+  formSelector: ".modal__form",
+  inputSelector: ".modal__input",
+  submitButtonSelector: ".modal__submit-btn",
+  inactiveButtonClass: "modal__submit-btn_disabled",
+  inputErrorClass: "modal__input_state_error",
+  errorClass: "modal__error",
+};
+
+const showInputError = (formElement, inputElement, errorMessage, config) => {
+  const errorMessageEl = formElement.querySelector(`#${inputElement.id}-error`);
+  errorMessageEl.textContent = errorMessage;
+  inputElement.classList.add(config.inputErrorClass);
+};
+
+const hideInputError = (formElement, inputElement, config) => {
+  const errorMessageEl = formElement.querySelector(`#${inputElement.id}-error`);
+  errorMessageEl.textContent = "";
+  inputElement.classList.remove(config.inputErrorClass);
+};
+
+const checkInputValidity = (formElement, inputElement, config) => {
+  if (!inputElement.validity.valid) {
+    showInputError(
+      formElement,
+      inputElement,
+      inputElement.validationMessage,
+      config
+    );
+  } else {
+    hideInputError(formElement, inputElement, config);
+  }
+};
+const hasInvalidInput = (inputList) => {
+  return inputList.some((input) => {
+    return !input.validity.valid;
+  });
+};
+
+const toggleButtonState = (inputList, buttonElement, config) => {
+  if (hasInvalidInput(inputList)) {
+    disabledButton(buttonElement, config);
+  } else {
+    buttonElement.disabled = false;
+    buttonElement.classList.remove(config.inactiveButtonClass);
+  }
+};
+
+const disabledButton = (buttonElement, config) => {
```

Comment:
* You have `disabledButton` inside `validation.js` which should be imported into `index.js`. You'll use it to **disable** the button after submission

Please, add `export ` here

```js
export const disabledButton = (buttonElement, config) => {
  buttonElement.disabled = true;
  buttonElement.classList.add(config.inactiveButtonClass);
};

```